File: app/src/main/java/de/luki2811/dev/inzidenzencheck/SettingsActivity.java
package de.luki2811.dev.inzidenzencheck;

import androidx.appcompat.app.AppCompatActivity;

import android.content.ActivityNotFoundException;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;
import android.widget.Switch;
import android.widget.Toast;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.File;

public class SettingsActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);

        File file = new File(getApplicationContext().getFilesDir(),MainActivity.fileNameSettings);
        Datein datei = new Datein(MainActivity.fileNameSettings);
        Switch settings_switch_0 = findViewById(R.id.settings_switch_0);
        if(file.exists()){
            try {
                JSONObject jsonObject = new JSONObject(datei.loadFromFile(this));
                settings_switch_0.setChecked(jsonObject.getBoolean("automaticPlaceInput"));
            } catch (JSONException e) {
                e.printStackTrace();
            }
        } else{
            JSONObject jsonObject = new JSONObject();
            try {
                jsonObject.put("automaticPlaceInput", Boolean.TRUE);
                datei.writeInFile(jsonObject.toString(), this);
                settings_switch_0.setChecked(true);
            } catch (JSONException e) {
                e.printStackTrace();
            }
        }

    }

    public void openBrowserToMyPage(View view){
        try {
            Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse("https://dev.luki2811.de/coronaInzidenzen/thirt-party-licenses.html"));
            startActivity(browserIntent);
        } catch (ActivityNotFoundException e) {
            Toast.makeText(this, "No application can handle this request." + " Please install a webbrowser",  Toast.LENGTH_LONG).show();
            e.printStackTrace();
        }
    }

    public void saveSettings(View view){
        Datein datein = new Datein(MainActivity.fileNameSettings);
        String oldJSON = datein.loadFromFile(this);

        Switch settings_switch_0 = findViewById(R.id.settings_switch_0);
        JSONObject JsonObject = null;
        try {
            JsonObject = new JSONObject(oldJSON);
            JsonObject.put("automaticPlaceInput",settings_switch_0.isChecked());
        } catch (JSONException e) {
            e.printStackTrace();
        }
        datein.writeInFile(JsonObject.toString(), this);

        Toast.makeText(this, getString(R.string.successfully_saved), Toast.LENGTH_SHORT).show();

        Intent intent = new Intent(this, MainActivity.class);
        startActivity(intent);
    }
}

File: app/src/main/java/de/luki2811/dev/inzidenzencheck/SettingsActivity.kt
package de.luki2811.dev.inzidenzencheck

import android.content.ActivityNotFoundException
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.switchmaterial.SwitchMaterial
import org.json.JSONException
import org.json.JSONObject
import java.io.File

class SettingsActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_settings)
        val file = File(applicationContext.filesDir, MainActivity.fileNameSettings)
        val datei = Datein(MainActivity.fileNameSettings)
        val settingsSwitchLastInput = findViewById<SwitchMaterial>(R.id.settings_switch_0)
        if (file.exists()) {
            try {
                val jsonObject = JSONObject(datei.loadFromFile(this))
                settingsSwitchLastInput.isChecked = jsonObject.getBoolean("automaticPlaceInput")
            } catch (e: JSONException) {
                e.printStackTrace()
            }
        } else {
            val jsonObject = JSONObject()
            try {
                jsonObject.put("automaticPlaceInput", true)
                datei.writeInFile(jsonObject.toString(), this)
                settingsSwitchLastInput.isChecked = true
            } catch (e: JSONException) {
                e.printStackTrace()
            }
        }
    }

    fun openBrowserToMyPage(view: View?) {
        try {
            val browserIntent = Intent(
                Intent.ACTION_VIEW,
                Uri.parse("https://dev.luki2811.de/coronaInzidenzen/thirt-party-licenses.html")
            )
            startActivity(browserIntent)
        } catch (e: ActivityNotFoundException) {
            Toast.makeText(
                this,
                "No application can handle this request." + " Please install a webbrowser",
                Toast.LENGTH_LONG
            ).show()
            e.printStackTrace()
        }
    }

    fun saveSettings(view: View?) {
        val datein = Datein(MainActivity.fileNameSettings)
        val oldJSON = datein.loadFromFile(this)
        val settingsSwitchLastInput = findViewById<SwitchMaterial>(R.id.settings_switch_0)
        var jsonObject: JSONObject? = null
        try {
            jsonObject = JSONObject(oldJSON)
            jsonObject.put("automaticPlaceInput", settingsSwitchLastInput.isChecked)
        } catch (e: JSONException) {
            e.printStackTrace()
        }
        datein.writeInFile(jsonObject.toString(), this)
        Toast.makeText(this, getString(R.string.successfully_saved), Toast.LENGTH_SHORT).show()
        val intent = Intent(this, MainActivity::class.java)
        startActivity(intent)
    }
}